app/models/king.rb

```diff
@@ -1,12 +1,18 @@
 class King < Piece
 
-
 	def valid_move?(new_x, new_y)
     	x_difference = (new_x.to_i - x_position).abs
     	y_difference = (new_y.to_i - y_position).abs
     	(x_difference <= 1) && (y_difference <= 1)
   	end
   
+  def valid_move_check(new_x, new_y)
+    if (off_board? super(new_x, new_y) && valid_move?(new_x, new_y))
```

Comment:
@bdbradley a few points:

1. I'd suggest to switch the methods `valid_move?` and `valid_move_check`. `valid_move?` should 'orchestrate' what is happening, calling super, etc... 
2. Following nr 1: `valid_move?` needs to be added to piece.rb as well so the call to super will make sense..
3. `off_board?`  needs arguments: `off_board?(new_x, new_y)`
4. off_board should be a part of super probably - since it is a general check and needs to be made on all pieces